arrow/util.py

```diff
@@ -23,6 +28,45 @@ def is_timestamp(value):
         return False
 
 
+def windows_datetime_from_timestamp(timestamp, tz=None):
+    """Computes datetime from timestamp. Supports negative timestamps on Windows platform."""
+    sec_frac, sec = math.modf(timestamp)
```

Comment:
My only concern is with the use of `modf`. It seems like it does some pecuiliar rounding:

```shell
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 16:52:21)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import math
>>> sec_frac, sec = math.modf(1002130120312.12388128318)
>>> sec_frac
0.1239013671875
>>> sec
1002130120312.0
```

Reply:
I don't think that's `modf` problem, but rather a python float64 precision problem. If you just store this number it will behave the same.

Reply:
I am reading about `modf` on StackOverflow and it seems to have some precision issues: https://stackoverflow.com/q/3886402/.

The post recommends using the decimal library instead:

```shell
❯ python3
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 16:52:21)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from decimal import Decimal
>>> dec = Decimal.from_float(1002130120312.12388128318)
>>> dec
Decimal('1002130120312.1239013671875')
>>> neg_dec = Decimal.from_float(-1002130120312.12388128318)
>>> neg_dec
Decimal('-1002130120312.1239013671875')
>>> float(dec % 1)
0.1239013671875
>>> float(neg_dec % 1)
-0.1239013671875
>>> dec.is_signed()
False
>>> neg_dec.is_signed()
True
```

Reply:
Ah I see. You are right, this seems to simply be due to the way that Python stores floats. It feels odd to me because I thought Python3 stores floats as a 64-bit double under the hood, so it seems odd that the precision is lost so early.

I played with a few examples, but this time I stored the large floating point in a variable rather than directly passing it in, and you are correct:
```shell
❯ python3
Python 3.7.3 (v3.7.3:ef4ec6ed12, Mar 25 2019, 16:52:21)
[Clang 6.0 (clang-600.0.57)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from decimal import Decimal
>>> import math
>>> test_dec = 1002130120312.12388128318
>>> test_dec
1002130120312.1239
>>> str(test_dec)
'1002130120312.1239'
>>> "{:f}".format(test_dec)
'1002130120312.123901'
>>> "{:.32f}".format(test_dec)
'1002130120312.12390136718750000000000000000000'
>>> test_dec = 312.12388128318
>>> test_dec
312.12388128318
>>> math.modf(test_dec)
(0.123881283180026, 312.0)
>>> Decimal.from_float(test_dec)
Decimal('312.123881283180026002810336649417877197265625')
>>> test_dec = 312.123881
>>> math.modf(test_dec)
(0.12388099999998303, 312.0)
>>> Decimal.from_float(test_dec)
Decimal('312.12388099999998303246684372425079345703125')
```

Since timestamps should generally only have 6 decimal places, this minimal loss of precision should be ok then.